inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContextUnitTest.java

```diff
@@ -0,0 +1,748 @@
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(eq("keyA"));
+            doReturn(false).when(propagation).isPropagatedDownWithinJVM(eq("keyB"));
+
+            InspectitContext ctxA = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxA.setData("keyA", "ctxA_valueA");
+            ctxA.setData("keyB", "ctxA_valueB");
+            ctxA.makeActive(true);
+
+            InspectitContext ctxB = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxB.setData("keyB", "ctxB_valueB");
+            ctxB.makeActive(true);
+
+            InspectitContext ctxC = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxC.makeActive(true);
+
+            assertThat(ctxA.getData("keyA")).isEqualTo("ctxA_valueA");
+            assertThat(ctxA.getData("keyB")).isEqualTo("ctxA_valueB");
+
+            assertThat(ctxB.getData("keyA")).isEqualTo("ctxA_valueA");
+            assertThat(ctxB.getData("keyB")).isEqualTo("ctxB_valueB");
+
+            assertThat(ctxC.getData("keyA")).isEqualTo("ctxA_valueA");
+            assertThat(ctxC.getData("keyB")).isNull();
+
+
+            ctxC.close();
+            ctxB.close();
+            //ensure no up propagation took place
+            assertThat(ctxA.getData("keyA")).isEqualTo("ctxA_valueA");
+            assertThat(ctxA.getData("keyB")).isEqualTo("ctxA_valueB");
+            ctxA.close();
+
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyNoMemoryLeakForAsyncCalls() {
+
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+
+            InspectitContext firstContext = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            firstContext.makeActive(true);
+
+            InspectitContext childContext = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            childContext.makeActive(true);
+            WeakReference<InspectitContext> firstContextWeak = new WeakReference<>(firstContext);
+
+            InspectitContext asyncContext = InspectitContext.createFromCurrent(commonTags, propagation, true);
+
+            childContext.close();
+            firstContext.close();
+            firstContext = null;
+
+            GcUtils.waitUntilCleared(firstContextWeak);
+
+            asyncContext.makeActive(true);
+            asyncContext.close();
+
+        }
+
+
+        @Test
+        void verifyDownPropagationForAsyncChildrenOnDifferentThreadWithRootActive() throws Exception {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(true).when(propagation).isPropagatedDownWithinJVM(any());
+
+            InspectitContext root = InspectitContext.createFromCurrent(commonTags, propagation, false);
+            root.setData("tag", "rootValue");
+            root.makeActive(false);
```

Comment:
when would we use this situation actually? is this `boolean` in the active only for testing purposes, if so you can have that on the package level and introduce `makeActive()` that calls it with `true`